decoder.js

```diff
@@ -136,8 +138,18 @@ GIFDecoder.prototype._readPalette = function() {
     
   this._palette = buf.slice(0, len);
   buf.consume(len);
-  
-  this._state = this._state === GIF_GLOBAL_PALETTE ? GIF_BLOCK : GIF_LZW;
+
+  if (this._state === GIF_GLOBAL_PALETTE) {
+    if (this._outputIndexed) {
+      this.format.palette = this._palette.slice(0);
```

Comment:
Er woops, I guess I'm getting my APIs mixed up. Slice should just return a shared memory view here since it's a Buffer. But should consider doing a copy operation re: this mutability issue